code/modules/projectiles/projectile/magic.dm

```diff
@@ -451,6 +451,142 @@
+
+/obj/item/projectile/magic/antimagic
+	name = "bolt of antimagic"
+	icon_state = "antimagic"
+
+/obj/item/projectile/magic/antimagic/on_hit(target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.anti_magic_check())
+			L.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		L.apply_status_effect(STATUS_EFFECT_ANTIMAGIC)
+
+/obj/item/projectile/magic/fetch
+	name = "bolt of fetching"
+	icon_state = "fetch"
+
+/obj/item/projectile/magic/fetch/on_hit(target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.anti_magic_check() || !firer)
+			L.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		var/atom/throw_target = get_edge_target_turf(L, get_dir(L, firer))
+		L.throw_at(throw_target, 200, 4)
+
+/obj/item/projectile/magic/sapping
+	name = "bolt of sapping"
+	icon_state = "sapping"
+
+/obj/item/projectile/magic/sapping/on_hit(target)
+	. = ..()
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, src, /datum/mood_event/sapped)
+
+/obj/item/projectile/magic/necropotence
+	name = "bolt of necropotence"
+	icon_state = "necropotence"
+
+/obj/item/projectile/magic/necropotence/on_hit(target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.anti_magic_check() || !L.mind || !L.mind.hasSoul)
+			L.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		to_chat(L, "<span class='danger'>Your body feels drained and there is a burning pain in your chest.</span>")
+		L.maxHealth -= 20
+		L.health = min(L.health, L.maxHealth)
+		if(L.maxHealth <= 0)
+			to_chat(L, "<span class='userdanger'>Your weakened soul is completely consumed by the [src]!</span>")
+			L.mind.hasSoul = FALSE
+		for(var/obj/effect/proc_holder/spell/spell in L.mind.spell_list)
+			spell.charge_counter = spell.charge_max
+			spell.recharging = FALSE
+			spell.update_icon()
+
+/obj/item/projectile/magic/wipe
+	name = "bolt of possession"
+	icon_state = "wipe"
+
+/obj/item/projectile/magic/wipe/on_hit(target)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		for(var/x in M.get_traumas())//checks to see if the victim is already going through possession
+			if(istype(x, /datum/brain_trauma/special/imaginary_friend/trapped_owner))
+				M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+				return BULLET_ACT_BLOCK
+		to_chat(M, "<span class='warning'>Your mind has been opened to possession!</span>")
+		possession_test(M)
+		return BULLET_ACT_BLOCK
```

Comment:
why are you returning block when it has hit at this point?

Reply:
good point